tools/functional-tester/etcd-runner/command/global.go

```diff
@@ -61,41 +56,41 @@ func newClient(eps []string, timeout time.Duration) *clientv3.Client {
 	return c
 }
 
-func doRounds(rcs []roundClient, rounds int) {
-	var mu sync.Mutex
-	var wg sync.WaitGroup
+func doRounds(rcs []roundClient, rounds int, requests int) {
+	var (
+		wg sync.WaitGroup
+	)
 
 	wg.Add(len(rcs))
 	finished := make(chan struct{})
+	limiter := rate.NewLimiter(rate.Limit(reqRate), reqRate)
 	for i := range rcs {
 		go func(rc *roundClient) {
 			defer wg.Done()
-			for rc.progress < rounds {
+			for rc.progress < rounds || rounds <= 0 {
+				if err := limiter.WaitN(context.Background(), requests/len(rcs)); err != nil {
+					log.Panicf("rate limiter error %v", err)
+				}
+
 				for rc.acquire() != nil { /* spin */
 				}
 
-				mu.Lock()
```

Comment:
Doesn't look safe:
1. P1 acquires/releases distributed mutex
2. P2 acquires distributed mutex
3. P1 sets count to 0.
4. P2 validates, sees count == 0.

Is there a good reason for removing these mutexes?